src/Tools/dotnet-monitor/Extensibility/ProgramExtension.cs

```diff
@@ -117,6 +134,14 @@ public async Task<EgressArtifactResult> EgressArtifact(ExtensionEgressPayload co
             _logger.ExtensionEgressPayloadCompleted(p.Id);
 
             EgressArtifactResult result = await parser.ReadResult();
+
+            p.Kill();
```

Comment:
Consider allowing the program to exit gracefully by using `Process.WaitForExitAsync`. You can use `CancellationTokenSource.CreateLinkedTokenSource` if you want to use a timeout as well so you don't indefinitely wait on it before attempting to kill it.

Reply:
Let me know if what I did looks right for this - I tried to reference some similar existing instances in our code.